### PR TITLE
M2

### PR DESCRIPTION
**Pull request by valeria1604**

### CODE REVIEW

File: src/com/company/M2/Task42.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+public class Task42 {
+    public static void main(String[] args) {
+        System.out.println("isInvisibleName([no name]): " + isInvisibleName(null));
+        System.out.println("isInvisibleName(\"\"): " + isInvisibleName(""));
+        System.out.println("isInvisibleName(\"\\n\"): " + isInvisibleName("\n"));
+        System.out.println("isInvisibleName(\"\\nMiag \"): " + isInvisibleName("\nMiag"));
+    }
+
+    public static boolean isInvisibleName(String name) {
+        return name == null || name.length() == 0 || name.equals(" ") || name.equals("\n") || name.equals("\t");
```
Comment (maxline) [inline src/com/company/M2/Task42.java]: you shouldn't hardore a values for check for invisible symbols. what the result method returns for string "   " (3 spaces), or "\n\t"?
read about String class methods isEmpty(), isBlank(), trim(). You can read more in the article, or google examples by yourself
https://www.baeldung.com/java-blank-empty-strings
use correct variant and add more different test cases in method main()

File: src/com/company/M2/Task44.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        System.out.println("2)[empty name]: " + makeNameClean(""));
+        System.out.println("3)bigar: " + makeNameClean("bigar"));
+        System.out.println("4)   ytuki: " + makeNameClean("   ytuki"));
+    }
+
+    public static String makeNameClean(String name) {
+        if (name == null || name.isBlank()) {
+            return "";
+        }
+        int amount = 0;
```
Comment (maxline) [inline src/com/company/M2/Task44.java]: name variable looks not matching enough. `amount` is more about of sum of numeric values.

File: src/com/company/M2/Task44.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        }
+        char[] chars = new char[amount];
+        int poz = 0;
+        for (int i = 0; i < name.length(); i++) {
+            if (Character.isLetter(name.charAt(i))) {
+                chars[poz] = name.charAt(i);
+                poz++;
+            }
+        }
+        return "CLEAN" + String.valueOf(chars) + "CLEAN";
```
Comment (maxline) [inline src/com/company/M2/Task44.java]: Let's leave this solution as varian1, and implent one more variant with using appropriate methods for trimming string from the String class methods.

File: src/com/company/M2/Task45.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        } else {
+            size = (name.length() - 1) / 2;
+        }
+        char[] chars = new char[size];
+        int poz = 0;
+        for (int i = 0; i < size; i++) {
+            chars[poz] = name.charAt(i);
+            poz++;
+        }
+        return String.valueOf(chars);
```
Comment (maxline) [inline src/com/company/M2/Task45.java]: Let's leave this solution as varian1, and implent one more variant with using appropriate methods for getting substring  from the String class methods.

File: src/com/company/M2/Task46.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+            if (name.charAt(i) == 'o') {
+                chars[i] = '4';
+            }
+            if (name.charAt(i) == 'a') {
+                chars[i] = '5';
+            } else {
+                chars[i] = name.charAt(i);
+            }
+        }
+        return "NOTFORYOU" + String.valueOf(chars) + "YESNOTFORYOU";
```
Comment (maxline) [inline src/com/company/M2/Task46.java]: Let's leave this solution as varian1, and implement one more variant with using appropriate methods for replacing substring from the String class methods.

File: src/com/company/M2/Task46.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    public static String decode(String name) {
+        char[] chars = new char[name.length() - 21];
```
Comment (maxline) [inline src/com/company/M2/Task46.java]: 21 is a magic number. all magic numbers should be a constant with explanation names

File: src/com/company/M2/Task46.java [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+        int poz = 0;
+        for (int i = 9; i < name.length() - 12; i++) {
```
Comment (maxline) [inline src/com/company/M2/Task46.java]: 9 and 12 is a magic numbers

**Review by maxline — CHANGES_REQUESTED** [review]
good job! let's improve the solution

File: src/com/company/M2/Task42.java (hunk 1 above)
Comment (valeria1604) [reply src/com/company/M2/Task42.java]: fixed

File: src/com/company/M2/Task44.java (hunk 2 above)
Comment (valeria1604) [reply src/com/company/M2/Task44.java]: fixed

File: src/com/company/M2/Task44.java (hunk 3 above)
Comment (valeria1604) [reply src/com/company/M2/Task44.java]: fixed

File: src/com/company/M2/Task45.java (hunk 4 above)
Comment (valeria1604) [reply src/com/company/M2/Task45.java]: fixed

File: src/com/company/M2/Task46.java (hunk 5 above)
Comment (valeria1604) [reply src/com/company/M2/Task46.java]: fixed

File: src/com/company/M2/Task46.java (hunk 6 above)
Comment (valeria1604) [reply src/com/company/M2/Task46.java]: fixed

File: src/com/company/M2/Task46.java (hunk 7 above)
Comment (valeria1604) [reply src/com/company/M2/Task46.java]: fixed

File: src/com/company/M2/Task46.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
+package com.company.M2;
+
+public class Task46 {
+    public static String firstWord = "NOTFORYOU";
+    public static String lastWord = "YESNOTFORYOU";
```
Comment (maxline) [inline src/com/company/M2/Task46.java]: please use correct naming for constants (`XXX_YYYY`). and do not forget `final` modifier

**Review by maxline — CHANGES_REQUESTED** [review]
good job! Just one small comment, take a look please

**Review by maxline — APPROVED** [review]
Approved

**Pull request merged**